sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -626,6 +626,7 @@ object DataSource extends Logging {
       serviceLoader.asScala.filter(_.shortName().equalsIgnoreCase(provider1)).toList match {
         // the provider format did not match any given registered aliases
         case Nil =>
+          val latestDocsURL = "https://spark.apache.org/docs/latest"
```

Comment:
I would actually avoid to leave the explicit doc link because we will have to fix it for every release. Just prose should be good enough.

Reply:
This is the link for the latest doc. I think it should be ok.

Reply:
I mean, if we happen to have Spark 3.0.0 then this link will be stale in 2.4.0.. no?

Reply:
The doc will be like 
https://github.com/apache/spark/pull/22121/files#diff-acdddc6cbd45ccd226bf151564b9cc40R11

It is about loading the module with `--package`

Reply:
But what if we happen to have more information specific to newer versions ..? For example, we could happen to have different group name rule, etc. in the future. Avoiding pointing out the latest shouldn't be too difficult .. You could just say "Please refer deployment section in Apache Avro data source guide.". if we happen to have other new changes specific to newer versions there, we should go down and fix the links in all the branches strictly in theory.

Reply:
Ah, make sense. Thank you!